ci_environment/mysql/templates/default/my.cnf.erb

```diff
@@ -51,14 +51,17 @@ basedir         = /usr
 datadir         = <%= node['mysql']['data_dir'] %>
 tmpdir          = /tmp
 skip-external-locking
+
 #
 # Instead of skip-networking the default is now to listen only on
 # localhost which is more compatible and is not less secure.
 bind-address            = <%= node['mysql']['bind_address'] %>
 
+innodb_file_per_table  = true
+innodb_file_format     = Barracuda
 collation_server       = utf8_unicode_ci
 character_set_server   = utf8
-
+innodb_large_prefix
```

Comment:
From the documentation you pointed out:
> When the innodb_large_prefix configuration option is enabled, this length limit is raised to 3072 bytes, for InnoDB tables that use the DYNAMIC and COMPRESSED row formats.

From http://dev.mysql.com/doc/refman/5.6/en/innodb-parameters.html#sysvar_innodb_large_prefix:

> Enable this option to allow index key prefixes longer than 767 bytes (up to 3072 bytes), for InnoDB tables that use the DYNAMIC and COMPRESSED row formats. (Creating such tables also requires the option values innodb_file_format=barracuda and innodb_file_per_table=true.) 

So yes we need `innodb_large_prefix` which needs `Barracuda` which needs `innodb_file_per_table=true`.